### PR TITLE
`astroquery.mast.Observations` get_cloud_uri(s) update

### PR DESCRIPTION
**Pull request by jaymedina**
Updating the `get_cloud_uri`/`s` methods to access the cloud anonymously using the inherited `enable_cloud_access` method from `MastQuerywithLogin`.

### CODE REVIEW

**Review by keflavich — APPROVED** [review]

**Comment by jaymedina** [comment]
I've just joined the astropy project so this should be the last time I need manual approval to run checks.

**Comment by jaymedina** [comment]
I see a message from Codecov saying `Added line #L756 was not covered by tests` , and the same for 758. Is this referring to the test_mast.py script under `tests/` ? How do I clear up this warning?

**Comment by bsipocz** [comment]
@jaymedina - we're not strict on codecov, as in many cases it reports nonsense, we mostly use it to monitor that test coverage is not dropping, in practice maintainers check the diff and those reports and ask for more tests if something is clearly not covered in a PR. 
But in this case, those lines are indeed not covered by either mock or remote tests. If you can think of a way to run tests for the cloud parts, without sharing credentials in a public repo, well, that would be a great addition.

(Btw, you can also run the remote tests locally, `pytest setup.py -R -P mast` is the command. There are a few currently failing for mast, totally unrelated to this PR, and as I see most only need an update into their asserts, or maybe rethink a logic of what we do check on. )

**Comment by bsipocz** [comment]
Ahh, OK, my bad. So, if anonymous access is indeed possible and free, then I would suggest adding a test that just does that. Originally I think it was the case I listed above that we should have handled credentials somehow.

**Comment by jaymedina** [comment]
OK thanks, and yes I can write up a test for this. I would put it in the `test_mast.py` under tests right?

**Comment by bsipocz** [comment]
> I would put it in the test_mast.py under tests right?

no, it should be in `astroquery/mast/tests/test_mast_remote.py`

**Comment by jaymedina** [comment]
Sounds good!

**Comment by jaymedina** [comment]
I'm getting an error when running my unit test on `pytest`: 

`ERROR: usage: pytest [options] [file_or_dir] [file_or_dir] [...]
pytest: error: unrecognized arguments: --doctest-rst
  inifile: /mypath/astroquery/setup.cfg
  rootdir: /mypath//astroquery`
  
  Do you recognize this error? @bsipocz

**Comment by bsipocz** [comment]
Make sure you have all the plugins installed and up-to-date (components of pytest-astropy). Also, note the typo I made in the command above, test via pytest may not work as easily, but do use ``python setup.py -R -P mast`` instead. 

Also, I would suggest to write a test function with working syntax before committing it, as it will take a lot of iterations and longer if you develop it directly in the codebase. (E.g. you can do it in an ipython session, or something similar).

File: astroquery/mast/observations.py
Comment (bsipocz) [inline astroquery/mast/observations.py:756]:
```suggestion
            raise RemoteServiceError('Please enable anonymous cloud access by calling `enable_cloud_dataset` method. '
                                     'See MAST Labs documentation for an example: https://mast-labs.stsci.io/#example-data-access-with-astroquery-observations')
```

File: astroquery/mast/observations.py
Comment (bsipocz) [inline astroquery/mast/observations.py:787]:
```suggestion
            raise RemoteServiceError('Please enable anonymous cloud access by calling `enable_cloud_dataset` method. '
                                     'See MAST Labs documentation for an example: https://mast-labs.stsci.io/#example-data-access-with-astroquery-observations')
```

File: astroquery/mast/tests/test_mast_remote.py
Comment (bsipocz) [inline astroquery/mast/tests/test_mast_remote.py:319]: We don't necessarily write assert messages in the tests, but you're right, they likely don't hurt to have :)

**Review by bsipocz — APPROVED** [review]

**Comment by bsipocz** [comment]
Thanks @jaymedina!

**Pull request merged**
